src/java/org/apache/cassandra/cql3/conditions/ColumnCondition.java

```diff
@@ -622,16 +855,21 @@ private static final class UDTFieldAccessBound extends Bound
         /**
          * The UDT field.
          */
+        @Nonnull
         private final FieldIdentifier field;
 
         /**
          * The conditions values.
          */
+        @Nonnull
         private final List<ByteBuffer> values;
 
         private UDTFieldAccessBound(ColumnMetadata column, FieldIdentifier field, Operator operator, List<ByteBuffer> values)
         {
             super(column, operator);
+            checkNotNull(field);
+            checkNotNull(values);
+            checkForUnsetValues(values);
             assert column.type.isUDT() && field != null;
```

Comment:
nit: `field` can't be `null` here anymore

Reply:
Removed the extra null check on field